sample-apps/java-sample-app/manual/src/main/java/software/amazon/adot/sampleapp/AwsXrayContextDataProvider.java

```diff
@@ -0,0 +1,32 @@
+package software.amazon.adot.sampleapp;
+
+        import io.opentelemetry.api.trace.Span;
+        import io.opentelemetry.api.trace.SpanContext;
+        import java.util.Collections;
+        import java.util.Map;
+        import org.apache.logging.log4j.core.util.ContextDataProvider;
+
+/**
+ * a format for consumption by AWS X-Ray and related services.
+ */
+public class AwsXrayContextDataProvider implements ContextDataProvider {
+    private static final String TRACE_ID_KEY = "AWS-XRAY-TRACE-ID";
+
+    @Override
+    public Map<String, String> supplyContextData() {
+        Span currentSpan = Span.current();
+        SpanContext spanContext = currentSpan.getSpanContext();
+        if (!spanContext.isValid()) {
+            return Collections.emptyMap();
+        }
+
+        String value =
+                "1-"
+                        + spanContext.getTraceId().substring(0, 8)
+                        + "-"
+                        + spanContext.getTraceId().substring(8)
```

Comment:
Shouldn't an x-ray trace id already be in this format? Where the only addition needed is the `@spanid`?

Reply:
Or does the trace id not contain the `1-` and `-`?

Reply:
This is converting to the xray format. `getTraceId` returns a OpenTelemetry trace id, which is in s different format according to the spec: https://opentelemetry.io/docs/reference/specification/trace/api/#retrieving-the-traceid-and-spanid

Reply:
Got it. Is this the official way to create x-ray trace ids or is this just a work around for the sake of an example? Does the upstream package have a way to use x-ray format traceids instead of OTel trace id format when starting new traces?

Reply:
to be clear, this piece of code is just converting from the Otel Trace id format to the Xray trace id format.

Reply:
Ahh I see. I mixed up the steps. I thought the ID generator would put the TraceID in x-ray format after doing the other steps for timestamp/random bits. TIL. Ty.